contrib/cirrus/runner.sh

```diff
@@ -407,8 +408,21 @@ dotest() {
         die "Found fallback podman '$fallback_podman' in \$PATH; tests require none, as a guarantee that we're testing the right binary."
     fi
 
+    # Catch invalid "TMPDIR == /tmp" assumptions; PR #19281
+    TMPDIR=$(mktemp --tmpdir -d CI_XXXX)
+    # tmp dir is commonly 1777 to allow all user to read/write
+    chmod 1777 $TMPDIR
+    export TMPDIR
+    fstype=$(findmnt -n -o FSTYPE --target $TMPDIR)
+    if [[ "$fstype" != "tmpfs" ]]; then
+        die "The CI test TMPDIR is not on a tmpfs mount, we need tmpfs to make the tests faster"
+    fi
+
```

Comment:
I'm assuming this is here and not `setup_environment.sh` because putting it there would be more complex?  Please consider if this is worth adding a comment about, should some future maintainer get the "bright" idea to try and move it :smile:

Reply:
I don't think it is easier one way or another I just went with the place where I know it will be called.
I guess if I move it it will effect all tests/tasks (not a bad thing but it could need more work to get them to pass)

Reply:
Gotcha.  No this is definitely the simplest place to do this.  Env. vars don't automatically pass through from `setup_environment.sh` -> `runner.sh`.  I think you'd also need some extra magic to ensure it also passed into the podman-in-podman environment.  I guess what I'm concerned about is if somebody thinks they can just copy-paste move this block elsewhere and it will magically work.  A comment could avoid them wasting their time.  How about:

```
# Note: Setting up a `$TMPDIR` elsewhere/for other tests should consider
# passthrough_envars(), `ssh $ROOTLESS_USER@localhost` (rootless)
# and/or podman-in-podman testing environments.
```